src/spox/_attributes.py

```diff
@@ -82,8 +93,11 @@ def __init__(self, concrete: Attr[T], outer_name: str):
     def copy(self) -> "_Ref[T]":
         return self
 
-    def _to_onnx(self, key: str) -> AttributeProto:
-        parent_type = self._concrete._to_onnx(key).type
+    def _to_onnx(self, key: str, only_typecheck: bool = False) -> AttributeProto:
+        parent_type = self._concrete._to_onnx(
+            key,
+            only_typecheck=True,  # TODO in this PR: make sure we never use the actual attribute proto.
```

Comment:
Does `only_typecheck=True` in this context look right to you, i.e. do we ever need to construct a protobuf with values in the `_Ref` case?

Reply:
`Ref` attributes are only used in the context of ONNX-functions which are very niche in idiomatic Spox usages. I'd be fine simply foregoing any optimization for them at this point in time.

Reply:
To be completely specific, a Ref attribute can only occur in a FunctionProto which defines a function. A function (when used as an operator) can itself have attributes and operators in its body can refer to them. That's what function attributes do. However, as far as I'm aware, since we never decided to fully support functions, I don't think this support is tested anywhere and this might be dead code.

Reply:
> Perhaps it's okay to remove this partial implementation if it makes the refactoring and optimisations easier?

It doesn't affect it, so I would leave it. Was just curious, thanks!